docs/spec.md

```diff
@@ -232,6 +236,42 @@ Informs the client about services that are no longer available. Only supported i
 }
 ```
 
+### Connection graph update
+
+Informs the client about updates to the connection graph. This is only sent to clients which have previously [subscribed to connection graph updates](#subscribe-connection-graph-updates). Only supported if the server previously declared that it has the `connectionGraph` [capability](#server-info).
+
+#### Fields
+
+- `op`: string `"connectionGraphUpdate"`
+- `topicUpdates`: array of newly added or updated topics
+  - `name`: string, topic name
+  - `publisherIds`: array of string, list of publisher IDs
+  - `subscriberIds`: array of string, list of subscriber IDs
+- `serviceUpdates`: array of newly added or updated services
+  - `name`: string, service name
+  - `providerIds`: array of string, list of service provider IDs
```

Comment:
I don't know if we should include this here, as services can in the end have only one provider...?

Reply:
For ROS that's true. I say leave it to support non-ROS systems with multiple providers for a single service endpoint.